packages/@lwc/integration-types/src/decorators/api.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { LightningElement, api } from 'lwc';
+
+// @ts-expect-error bare decorator cannot be used
+api();
+
+// @ts-expect-error decorator doesn't work on classes
+@api
+export default class Test extends LightningElement {
+    @api optionalProperty?: string;
+    @api propertyWithDefault = true;
+    @api nonNullAssertedProperty!: object;
+    @api method() {}
+    @api getter(): undefined {}
+    @api setter(_: string) {}
+}
+
+// @ts-expect-error decorator doesn't work on non-component classes
+@api
+export class NonComponent {
+    // @ts-expect-error decorator doesn't work on non-component classes
+    @api optionalProperty?: string;
+    // @ts-expect-error decorator doesn't work on non-component classes
+    @api propertyWithDefault = true;
+    // @ts-expect-error decorator doesn't work on non-component classes
+    @api nonNullAssertedProperty!: object;
+    // @ts-expect-error decorator doesn't work on non-component classes
+    @api method() {}
+    // @ts-expect-error decorator doesn't work on non-component classes
+    @api getter(): undefined {}
+    // @ts-expect-error decorator doesn't work on non-component classes
+    @api setter(_: string) {}
+}
```

Comment:
As far as I know, there's no way to detect via the type system whether any given class extends an unknown superclass. But TypeScript will always know whether or not the superclass is a `LightningElement`<sup>1</sup>. So I think we can match the runtime behavior -- is decorating a non-`LightningElement` valid at runtime?

<sup>1</sup> I'll need to add tests for a superclass typed as `any` to validate that case.

Reply:
Decorating a non-LightningElement [compiles successfully](https://stackblitz.com/edit/salesforce-lwc-c3p9ux?file=src%2Fmodules%2Fx%2Fapp%2Frando.js&title=LWC%20playground) (and applies `registerDecorators()` if the class `extends` something), but does nothing at runtime.

What I meant to say is that our TypeScript types should be stricter than the current compiler – it can give a red squiggly for an `@api`/`@track`/`@wire` when used on a class that extends a non-`LightningElement`.